### ISSUE TITLE
Copying UTF-8 is messed up

### ISSUE THREAD

**Issue by bdcaf** [neovim/neovim#11432]
I noticed an annoying behavior pasting german text from nvim to other apps. All umlauts appear to become two characters.  At first I expected my terminal or terminal emulator to be at fault, but by chance I opened regular vim and noticed it behaves as expected.

- `nvim --version`:
```
NVIM v0.4.3
Build type: Release
LuaJIT 2.0.5
```
- `vim -u DEFAULTS` (version: ) behaves differently? 
  no
- Operating system/version:
MacOS 10.15.1

- Terminal name/version:
- `$TERM`: Terminal.app, iTerm
- encoding and fileencoding are UTF-8

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
```
insert text: Müller (ü is typed <M-u> <u>)
```
iMüller    
```
Copy to System clipboar
```
<Esc>
"+yy
```

### Actual behaviour

Pasting into e.g. notepad results: `M√ºller`

### Expected behaviour

Pasting into e.g. notepad results: `Müller`.
Vim 8.1 has this behavior.

**Comment by on3iro** [comment]
Same issue here

**Comment by bfredl** [comment]
what does "locale" show? (in the same term as you would run vim). Maybe `pbcopy` misbehaves if locale encoding is wrong.

**Comment by bdcaf** [comment]
in fish (my default terminal): 
```
LANG="de_AT.UTF-8"
LC_COLLATE="en_GB.UTF-8"
LC_CTYPE="en_GB.UTF-8"
LC_MESSAGES="en_GB.UTF-8"
LC_MONETARY="en_GB.UTF-8"
LC_NUMERIC="en_GB.UTF-8"
LC_TIME="en_GB.UTF-8"
LC_ALL="en_GB.UTF-8"
```

in nvim terminal (uses zsh as default) also:
```
LANG="en_GB"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```

**Comment by bfredl** [comment]
`LANG="en_GB"` seems wrong, it should be `LANG="en_GB.UTF-8"` or anything that ends with "UTF-8".

**Comment by bdcaf** [comment]
Yeah it got me to read through my dotfiles cause I don‘t recall setting it. Though I doubt it‘s the cause as it isn‘t affecting vim.

**Comment by justinmk** [comment]
> Though I doubt it‘s the cause as it isn‘t affecting vim.

Vim doesn't use pbcopy/pbpaste for clipboard, so that isn't relevant. Did you try in Nvim after fixing the env vars? Also what does this print in your shell (outside of Nvim):

    pbpaste

**Comment by bdcaf** [comment]
Still haven’t found where zsh gets the bad config. Not sure why zsh inside nvim has different environment. Could you give a set of vim commands to set a working LANG?

**Comment by bdcaf** [comment]
That might be it - I did: `:let $LANG='en_US.UTF-8'` in nvim and the copying did work. Still haven't located the source of the bad setting. Wonder if you might have an idea.

**Comment by bdcaf** [comment]
I might have the culprit - `:language`.

In `:help :language` is:
```
Some systems accept aliases like "en" or "en_US",
```

Applarently vim accepts it while nvim doesn't.

**Comment by justinmk** [comment]
I am not sure why we let env vars ever set a non-UTF8 locale, in the first place. 

see also https://github.com/mpv-player/mpv/commit/1e70e82baa9193f6f027338b0fab0f5078971fbe

i wonder if [this comment](https://github.com/mpv-player/mpv/commit/1e70e82baa9193f6f027338b0fab0f5078971fbe) is useful:

>     mpa->locale = newlocale(LC_ALL_MASK - LC_CTYPE_MASK, "C", (locale_t)0);
>
> This will make a locale object that's "C" (guaranteed to exist; won't fail except for possible OOM) in all categories by LC_CTYPE, and matches the default locale (determined by environment or system default) in LC_CTYPE.

**Comment by okmanideep** [comment]
`:echo $LANG` outputs `en_IN.UTF-8`

but 👍 is copied as üëç even though it is displayed properly in nvim
